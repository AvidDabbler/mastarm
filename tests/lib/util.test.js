/* globals describe, expect, it, jest */

const mockSyncFn = jest.fn(() => { return { status: 0, depsWereOk: true } })
jest.mock('check-dependencies', () => { return { sync: mockSyncFn } })

const util = require('../../lib/util')

describe('util.js', () => {
  describe('getCssFiles', () => {
    it('should return correct results', () => {
      expect(util.getCssFiles({ css: false })).toMatchSnapshot()
      expect(util.getCssFiles({ css: true })).toMatchSnapshot()
      expect(util.getCssFiles({ css: true, cssFiles: 'a:b c' })).toMatchSnapshot()
    })
  })

  it('makeGetFn should find the right value', () => {
    expect(util.makeGetFn([{ a: 1 }, { a: 2 }])('a')).toEqual(1)
  })

  describe('parseEntries', () => {
    it('should return default file paths', () => {
      expect(util.parseEntries([], () => null)).toMatchSnapshot()
    })

    it('should return inputted file paths', () => {
      expect(util.parseEntries(['a:b', 'c'])).toMatchSnapshot()
    })
  })

  it('popMastarmFromArgv should remove the mastarm command', () => {
    const argv1 = process.argv[1]
    expect(argv1).toContain('mastarm')
    util.popMastarmFromArgv()
    expect(process.argv[1]).not.toBe(argv1)
  })

  it('updateDependencies should run check-dependencies package', () => {
    util.updateDependencies()
    expect(mockSyncFn).toBeCalled()
  })
})
