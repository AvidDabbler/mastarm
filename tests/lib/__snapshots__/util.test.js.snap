exports[`util.js getCssFiles should return correct results 1`] = `Array []`;

exports[`util.js getCssFiles should return correct results 2`] = `
Array [
  "lib/styles.css:assets/index.css"
]
`;

exports[`util.js getCssFiles should return correct results 3`] = `
Array [
  "a:b",
  "c"
]
`;

exports[`util.js parseEntries should return default file paths 1`] = `
Array [
  Array [
    "bin/mastarm",
    "assets/index.js"
  ]
]
`;

exports[`util.js parseEntries should return inputted file paths 1`] = `
Array [
  Array [
    "a",
    "b"
  ],
  Array [
    "c",
    "assets/c"
  ]
]
`;
